effectful/ops/syntax.py

```diff
@@ -1164,6 +1140,39 @@ def syntactic_eq[T](x: Expr[T], other: Expr[T]) -> bool:
         return x == other
 
 
+@syntactic_eq.register
+def _(x: Term, other) -> bool:
+    if not isinstance(other, Term):
```

Comment:
I think the case `isinstance(other, Term) and not isinstance(x, Term)` is now missing

Reply:
This is handled by the base case `x == other`.